http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/HTTPAppDescriptorDAO.scala

```diff
@@ -0,0 +1,78 @@
+package org.broadinstitute.dsde.workbench.leonardo
+package dao
+
+import cats.effect.Effect
+import cats.implicits._
+import cats.mtl.Ask
+import io.chrisdavenport.log4cats.StructuredLogger
+import io.circe.Decoder
+import io.circe.yaml.parser
+import org.broadinstitute.dsde.workbench.model.TraceId
+import org.broadinstitute.dsde.workbench.leonardo.JsonCodec._
+import org.broadinstitute.dsde.workbench.leonardo.dao.HttpAppDescriptorDAO._
+import org.http4s.{Method, Request, Response, Uri}
+import org.http4s.client.Client
+
+class HttpAppDescriptorDAO[F[_]](httpClient: Client[F])(implicit logger: StructuredLogger[F], F: Effect[F])
+    extends AppDescriptorDAO[F] {
+  override def getDescriptor(path: String)(implicit ev: Ask[F, TraceId]): F[AppDescriptor] =
+    for {
+      _ <- logger.info(s"Attempting to retrieve descriptor $path")
+      yamlStr <- resolveUri(path)
+      _ <- logger.info(s"Attempting to parse descriptor for $path as a yaml string into JSON...")
+      json <- F.fromEither(parser.parse(yamlStr))
+      _ <- logger.info(s"Attempting to parse json for $path into AppDescriptor model...")
+      res <- F.fromEither(json.as[AppDescriptor])
+      _ <- logger.info(s"Succesfully processed descriptor at $path")
+    } yield res
+
+  private[dao] def resolveUri(path: String)(implicit ev: Ask[F, TraceId]): F[String] =
+    for {
+      resp <- httpClient.expectOr[String](
+        Request[F](
+          method = Method.GET,
+          uri = Uri.unsafeFromString(path)
+        )
+      )(onError(path))
+    } yield resp
+
+  private def onError(path: String)(response: Response[F])(implicit ev: Ask[F, TraceId]): F[Throwable] =
+    for {
+      traceId <- ev.ask
+      body <- response.bodyText.compile.foldMonoid
+      _ <- logger.error(s"${traceId} | call failed: $body")
+    } yield AppDescriptorException(traceId, path, body)
+}
+object HttpAppDescriptorDAO {
+  implicit val appDescriptorDecoder: Decoder[AppDescriptor] = Decoder.instance { d =>
+    for {
+      name <- d.downField("name").as[String]
+      author <- d.downField("author").as[String]
+      description <- d.downField("description").as[String]
+      version <- d.downField("version").as[String]
+      services <- d.downField("services").as[Map[String, CustomAppService]]
+    } yield AppDescriptor(name, author, description, version, services)
+  }
+
+  implicit val customAppServiceDecoder: Decoder[CustomAppService] = Decoder.instance { d =>
+    for {
+      image <- d.downField("image").as[ContainerImage]
+      port <- d.downField("port").as[Option[Int]]
```

Comment:
I think `port` is required (prob shouldn't default to 80 -- 80 may not work anyway due to ingress)